libsolidity/analysis/ViewPureChecker.cpp

```diff
@@ -116,31 +116,23 @@ class AssemblyViewPureChecker: public boost::static_visitor<void>
 
 bool ViewPureChecker::check()
 {
-	// The bool means "enforce view with errors".
-	map<ContractDefinition const*, bool> contracts;
+	vector<ContractDefinition const*> contracts;
 
 	for (auto const& node: m_ast)
 	{
 		SourceUnit const* source = dynamic_cast<SourceUnit const*>(node.get());
 		solAssert(source, "");
-		bool enforceView = source->annotation().experimentalFeatures.count(ExperimentalFeature::V050);
-		for (ContractDefinition const* c: source->filteredNodes<ContractDefinition>(source->nodes()))
-			contracts[c] = enforceView;
+		auto const& sourceContracts = source->filteredNodes<ContractDefinition>(source->nodes());
+		contracts.insert(contracts.end(), sourceContracts.begin(), sourceContracts.end());
```

Comment:
Can you use `contracts += source->filteredNodes<ContractDefinition>(source->nodes())`?

Reply:
Didn't know that that is defined for std::vector in libdevcore - I'll change it.